roles/openshift_node/tasks/main.yml

```diff
@@ -63,11 +63,15 @@
   lineinfile:
     dest: /etc/sysconfig/docker
     regexp: '^OPTIONS=.*'
-    line: "OPTIONS='--insecure-registry={{ openshift.node.portal_net }} --selinux-enabled'"
+    line: "OPTIONS='--insecure-registry={{ openshift.node.portal_net }} \
+{% if ( ansible_selinux and ansible_selinux.status == '''enabled''' ) or
+      deployment_type == '''enterprise''' %}--selinux-enabled{% endif %}'"
```

Comment:
Instead of doing this here, it may make more sense to use the fail task instead:
```
- fail:
    msg: "SELinux is disabled, This deployment type requires that SELinux is enabled."
    when: (not ansible_selinux or ansible_selinux.status != 'enabled') and deployment_type in ['enterprise', 'online']
```

Then the tasks within roles/openshift_node/tasks/main.yml wouldn't need to test for deployment_type.